kgforge/specializations/models/rdf/directory_service.py

```diff
@@ -61,30 +64,43 @@ def resolve_context(self, iri: str) -> Dict:
     def generate_context(self) -> Dict:
         return self._generate_context()
 
-    def _build_shapes_map(self) -> Dict:
-        query = """
-            PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
-            PREFIX sh: <http://www.w3.org/ns/shacl#>
-            SELECT ?type ?shape WHERE {
-                { ?shape sh:targetClass ?type .}
-                UNION {
-                    SELECT (?shape as ?type) ?shape WHERE {
-                        ?shape a sh:NodeShape .
-                        ?shape a rdfs:Class
-                    }
-                }
-            } ORDER BY ?type"""
+    def _build_shapes_map(self) -> Tuple[Dict, Dict, Dict]:
+        query = build_shacl_query(
+            defining_property_uri=self.NXV.shapes,
+            deprecated_property_uri=OWL.deprecated,
+            deprecated_optional=True,
+            search_in_graph=True,
+            context=self.context,
+        )
         res = self._graph.query(query)
-        return {row["type"]: row["shape"] for row in res}
+        class_to_shape = {}
+        shape_to_defining_resource = {}
+        defining_resource_to_named_graph = {}
+        for row in res:
```

Comment:
I'm making an assumption here, and I'm not sure, 
But I think inside the SPARQLResult object you iterate over objects of type ResultRow, and because of the shape GeoShape which doesn't have a type, I think none of the result rows have a type (to keep them all uniform, only the intersection of the columns are available), so if "type" in row is always false. 

However if you do
row = row.asdict()
Then "type" in row is True for all shapes but GeoShape.

Reply:
In fact I wouldn't `if "type" in row`, I would do a regular access surrounded by a try-except block, and maybe raise a custom exception from this.

That is, if "type" is always to be expected from a row (that seems to be the reason by the tests are failing). If "type" is not always expected, then discard this comment

Reply:
I go with `if row["type"]` which work/